js/projectHandler.js

```diff
@@ -0,0 +1,129 @@
+$(document).ready(function() {
+  project.populateFilters();
+  project.filterHandler();
+  project.readmoreHandler();
+  project.navHandler();
+  project.readMoreButton();
+});
+
+var project = {};
+project.populateFilters = function() {
+  $('.portfolio').find('.col-12').each(function() {
+
+      //grab the types form data-type
+      var types = $(this).find('.type').attr('data-type');
+
+      // console.log(types);
+      //split them into an array
+      var typeArray = types.split("&");
+
+      //for each item, create a option tag.
+      typeArray.forEach(function(item, index) {
+        //remove space
+        item = item.trim();
+        //if the array is not empty and not exist in the options then make a new one
+        if (item !== '' && $('#category-type option[value="' + item + '"]').length === 0) {
+          var optionTag = '<option value="' + item + '">' + item + '</option>';
+          $('#category-type').append(optionTag);
+        }
+      });
+  });
+};
+
+// display the related project tag by searching optionValue inside tagValue
+// for example search "css" in "css & html"
+project.filterHandler = function() {
+    $('#category-type').on('change', function() {
+      var optionValue = $(this).val();
+      if ($(this).val()) {
+        $('.portfolio .col-12').hide();
+        $('.portfolio').find('.col-12').each(function() {
+          var thisTag = $(this);
+          var tagValue = $(this).find('.type').attr('data-type');
+          console.log(tagValue);
+          if (tagValue.search(optionValue) != -1) {
+            thisTag.fadeIn();
+          };
+        });
+      } else {
+        $('.portfolio .col-12').fadeIn();
+        $('.template').hide();
+      }
+      //  $('#category-type').val('');
+    });
+  }
+
+  // insert someting after index
+  String.prototype.insert = function(index, string) {
+    if (index > 0)
+      return this.substring(0, index) + string + this.substring(index, this.length);
+    else
+      return string + this;
+  };
+
+  project.readmoreHandler = function() {
+    $('.portfolio').find('.col-12 .content').each(function() {
+      var content = $(this).text();
+      var wordCount = 0;
+
+      //count 100 word and hide all the contents after it.
+      for (var j = 0; j < content.length; j++) {
+        if (content[j] == " ") {
+          wordCount = wordCount + 1;
+          if (wordCount == 100) {
+            //insert the prewrite hide class
+            content = content.insert(j, "<span class='hide'>");
+            content = content.insert(content.length, "</span>");
+            content = content.insert(content.length, "<p class='more-button'> ...more</p>");
+            $(this).html(content).contents();
+          }
+        }
+      }
+    });
+  }
+project.readMoreButton = function(){
+  $('.content').on('click', '.more-button', function() {
```

Comment:
Is it readmore (line 64) or readMore (line 84)? Keep your capitalization style consistent for easier reading.